src/neovimconnector.cpp

```diff
@@ -195,12 +195,12 @@ NeovimConnector* NeovimConnector::spawn(const QStringList& params, const QString
 {
 	QProcess *p = new QProcess();
 	QStringList args;
+	// Neovim accepts a `--' argument after which only filenames are passed.
+	// If the user has supplied it, our arguments must appear before.
```

Comment:
This comment is definitely true, and if something like this was being constructed before:

    nvim -- --embed --headless

Then nvim should have opened files literally named `--embed` `--headless`.